code/controllers/subsystem/explosions.dm

```diff
@@ -0,0 +1,169 @@
+		var/list/currentrun = src.currentrun
+		while(currentrun.len)
+			var/atom/thing = currentrun[currentrun.len]
+			currentrun.len--
+			if(thing)
+				var/turf/T = thing
+				T.explosion_level = max(T.explosion_level, EXPLODE_LIGHT)
+				T.ex_act(EXPLODE_LIGHT)
+				lowturf.Remove(thing)
+			else
+				lowturf.Remove(thing)
+		cost_lowturf = MC_AVERAGE(cost_lowturf, TICK_DELTA_TO_MS(TICK_USAGE_REAL - timer))
+		src.currentrun = medturf.Copy()
+		//cache for sanic speed (lists are references anyways)
+		currentrun = src.currentrun
+		while(currentrun.len)
+			var/atom/thing = currentrun[currentrun.len]
+			currentrun.len--
+			if(thing)
+				var/turf/T = thing
+				T.explosion_level = max(T.explosion_level, EXPLODE_HEAVY)
+				T.ex_act(EXPLODE_HEAVY)
+				medturf.Remove(thing)
+			else
+				medturf.Remove(thing)
+		cost_medturf = MC_AVERAGE(cost_medturf, TICK_DELTA_TO_MS(TICK_USAGE_REAL - timer))
+		src.currentrun = highturf.Copy()
+		//cache for sanic speed (lists are references anyways)
+		currentrun = src.currentrun
+		while(currentrun.len)
+			var/atom/thing = currentrun[currentrun.len]
+			currentrun.len--
+			if(thing)
+				var/turf/T = thing
+				T.explosion_level = max(T.explosion_level, EXPLODE_DEVASTATE)
+				T.ex_act(EXPLODE_DEVASTATE)
+				highturf.Remove(thing)
+			else
+				highturf.Remove(thing)
+		cost_highturf = MC_AVERAGE(cost_highturf, TICK_DELTA_TO_MS(TICK_USAGE_REAL - timer))
+
+
+		if(state != SS_RUNNING)
+			return
+		resumed = 0
+		currentpart = SSEXPLOSIONS_OBJECTS
+
+	if(currentpart == SSEXPLOSIONS_OBJECTS)
+		src.currentrun = highobj.Copy()
+		//cache for sanic speed (lists are references anyways)
+		var/list/currentrun = src.currentrun
+		while(currentrun.len)
+			var/atom/thing = currentrun[currentrun.len]
+			currentrun.len--
+			if(thing)
+				var/obj/O = thing
+				O.ex_act(EXPLODE_DEVASTATE)
+				highobj.Remove(thing)
+			else
+				highobj.Remove(thing)
+		cost_highobj = MC_AVERAGE(cost_highobj, TICK_DELTA_TO_MS(TICK_USAGE_REAL - timer))
+		src.currentrun = medobj.Copy()
+		//cache for sanic speed (lists are references anyways)
+		currentrun = src.currentrun
+		while(currentrun.len)
+			var/atom/thing = currentrun[currentrun.len]
+			currentrun.len--
+			if(thing)
+				var/obj/O = thing
+				O.ex_act(EXPLODE_HEAVY)
+				medobj.Remove(thing)
+			else
+				medobj.Remove(thing)
+		cost_medobj = MC_AVERAGE(cost_medobj, TICK_DELTA_TO_MS(TICK_USAGE_REAL - timer))
+		src.currentrun = lowobj.Copy()
+		//cache for sanic speed (lists are references anyways)
+		currentrun = src.currentrun
+		while(currentrun.len)
+			var/atom/thing = currentrun[currentrun.len]
+			currentrun.len--
+			if(thing)
+				var/obj/O = thing
+				O.ex_act(EXPLODE_LIGHT)
+				lowobj.Remove(thing)
+			else
+				lowobj.Remove(thing)
+		cost_lowobj = MC_AVERAGE(cost_lowobj, TICK_DELTA_TO_MS(TICK_USAGE_REAL - timer))
+		if(state != SS_RUNNING)
+			return
+		resumed = 0
+		currentpart = SSEXPLOSIONS_TURFS
+	currentpart = SSEXPLOSIONS_TURFS
```

Comment:
need newline here for The Travis CI build

Reply:
right, thanks